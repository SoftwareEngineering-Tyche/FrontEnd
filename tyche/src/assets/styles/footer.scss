.footer {
    padding: 2% 16%;
    @media (max-width: 768px) {
        padding: 4%;
    }
    background-color: lightgray;
    .options {
        width: 100%;
        display: flex;
        justify-content: space-between;
        color: darkslategray;
        margin-bottom: 10px;
        @media (max-width: 768px) {
            justify-content: space-around;
        }
        .option {
            display: flex;
            justify-content: center;
            align-items: center;
            flex-direction: column;
            text-align: center;
            margin: 8px 0;
            .icon-container {
                width: 36px;
                height: 36px;
                .icon {
                    max-width: 40px;
                    max-height: 40px;
                }
            }
        }
    }
    .actions-container {
        display: flex;
        justify-content: space-between;
        margin-top: 24px;
        @media (max-width: 768px) {
            display: block;
        }
        .contact {
            margin-top: 16px;
            .contact-info {
                display: flex;
                align-items: center;
                .email-icon {
                    margin-left: 8px;
                }
                .email-address {
                    text-decoration: none;
                    color: black;
                    :hover {
                        color: #CDBDFF;
                    }
                }
            }
        }
        .tyche-info {
            @media (min-width: 768px) {
                max-width: 30%;
            }
            @media (max-width: 768px) {
                width: 100%;
                margin-bottom: 16px;
            }
            .title {
                display: flex;
                align-items: center;
                .title-txt {
                    margin-right: 8px;
                    font-size: 20px;
                    font-weight: 400;
                }
            }
            .sub-title {
                color: gray;
            }
        }
        .user-email {
            .email-form {
                display: flex;
                align-items: center;
                margin-top: 8px;
                width: 100%;
                .email-input {
                    background-color: transparent;
                    border-radius: 8px;
                    border: 1px solid #2F3A8F;
                    padding: 7px;
                    width: 100%;
                }
                .btn {
                    background-color: #CDBDFF;
                    color: #2F3A8F;
                    margin-right: 8px;
                    border-radius: 8px;
                }
                .disabled-btn {
                    background-color: darkgray;
                    color: gray;
                }
            }
        }
        .services-container {
            @media (max-width: 768px) {
                width: 100%;
                margin-bottom: 16px;
            }
            .services {
                display: flex;
                flex-direction: column;
                .service {
                    color: gray;
                    font-size: 14px;
                    cursor: pointer;
                }
                .service:hover {
                    color: #2F3A8F;
                }
            }
        }
    }
    .copy-right {
        margin-top: 24px;
        font-size: 14px;
        color: #2F3A8F;
    }
}