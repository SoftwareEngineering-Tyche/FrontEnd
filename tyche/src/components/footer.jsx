import React, { memo, useState } from "react";
import "../assets/styles/footer.scss";
import Button from '@mui/material/Button';
import walletIcon from '../assets/images/wallet.svg';
import collectionsIcon from '../assets/images/collection.svg';
import nftIcon from '../assets/images/nft.svg';
import saleIcon from '../assets/images/sale.svg';
import EmailRoundedIcon from '@mui/icons-material/EmailRounded';
import Snackbar from '@mui/material/Snackbar';
import MuiAlert from '@mui/material/Alert';
import imageSample from "../assets/images/image.png";
import { Divider } from "@mui/material";
import Link from '@mui/material/Link';

const Alert = React.forwardRef(function Alert(props, ref) {
    return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
});

function Footer() {

    const [email, setEmail] = useState(undefined);
    const [isEmailValid, setIsEmailValid] = useState(true);
    const [pressSubmit, setPressSubmit] = useState(false);

    const emailRegex = RegExp(/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/)

    const checkEmail = (email) => {
        const res = emailRegex.test(email)
        setIsEmailValid(res);
    }

    const getOptions1 = () => {
        return (<>
            <div className="option">
                <div className="icon-container"><img src={walletIcon} className="icon" /></div>
                <span>دسترسی مستقیم به کیف پول</span>
            </div>
            <div className="option">
                <div className="icon-container"><img src={collectionsIcon} className="icon" /></div>
                <span>ساخت مجموعه‌های متنوع</span>
            </div>
        </>);
    }
    const getOptions2 = () => {
        return (<>
            <div className="option">
                <div className="icon-container"><img src={nftIcon} className="icon" /></div>
                <span>اضافه کردن آثار دلخواه</span>
            </div>
            <div className="option">
                <div className="icon-container"><img src={saleIcon} className="icon" /></div>
                <span>لیست کردن آثار برای فروش</span>
            </div>
        </>);
    }
    const getCustomerServices = () => {
        return (
            <div className="services-container">
                <div className="service-title">خدمات مشتریان</div>
                <div className="services">
                    <span className="service">پاسخ به پرسش‌های متداول</span>
                    <span className="service">حریم خصوصی</span>
                    <span className="service">شرایط استفاده</span>
                    <span className="service">گزارش باگ</span>
                </div>
            </div>
        );
    }
    const getMyAccountServices = () => {
        return (
            <div className="services-container">
                <div className="service-title">حساب کاربری من</div>
                <div className="services">
                    <Link href='/profile' underline='none' color="inherit">
                        <span className="service">پروفایل</span>
                    </Link>
                    <Link href='/profile/favorites' underline='none' color="inherit">
                        <span className="service">علاقه‌مندی‌‌ها</span>
                    </Link>
                    <Link href='/profile/collections' underline='none' color="inherit">
                        <span className="service">دارایی‌ها</span>
                    </Link>
                    <Link href='/profile/creations' underline='none' color="inherit">
                        <span className="service">ساخته شده‌ها</span>
                    </Link>
                </div>
            </div>
        );
    }
    return (
        <div className="footer">
            <div className="options">
                {window.innerWidth <= 768 ? <>
                    <div>{getOptions1()}</div>
                    <div>{getOptions2()}</div>
                </>
                    : <> {getOptions1()} {getOptions2()} </>}
            </div>
            <Divider />
            <div className="actions-container">
                <div className="tyche-info">
                    <div className="title">
                        <img src={imageSample} width={36} height={36} />
                        <span className="title-txt">تایکی</span>
                    </div>
                    <div className="sub-title">
                        اولین و بزرگ‌ترین پلتفرم ایرانی در حوزه توکن‌های غیر قابل تعویض (NFT) برای جمع‌آوری، خرید، فروش و جستجو در اقلام دیجیتال منحصر به فرد
                    </div>
                </div>
                {window.innerWidth <= 768 ? <><div style={{ display: 'flex' }}>{getMyAccountServices()} {getCustomerServices()}</div></>
                    : <> {getMyAccountServices()} {getCustomerServices()} </>}
                <div>
                    <div className="user-email">
                        <div className="title">از آخرین اخبار تایکی مطلع شوید</div>
                        <div className="email-form">
                            <input
                                type="text"
                                placeholder="ایمیل خود را وارد کنید"
                                value={email}
                                onChange={event => { setEmail(event.target.value); checkEmail(event.target.value); }}
                                className="email-input"
                            />
                            <Button classes={{ root: 'btn', disabled: 'disabled-btn' }} disabled={!email} onClick={() => setPressSubmit(true)}>ثبت</Button>
                            {isEmailValid && <Snackbar open={pressSubmit} autoHideDuration={3000} onClose={() => setPressSubmit(false)}>
                                <Alert onClose={() => setPressSubmit(false)} severity="success" sx={{ width: '100%' }}>ایمیل شما با موفقیت ثبت شد!</Alert>
                            </Snackbar>}
                            {!isEmailValid && <Snackbar open={pressSubmit} autoHideDuration={3000} onClose={() => setPressSubmit(false)}>
                                <Alert onClose={() => setPressSubmit(false)} severity="error" sx={{ width: '100%' }}>آدرس ایمیل وارد شده نامعتبر است!</Alert>
                            </Snackbar>}
                        </div>
                    </div>
                    <div className="contact">
                        <span className="title">ارتباط با ما</span>
                        <div className="contact-info">
                            <EmailRoundedIcon classes={{ root: "email-icon" }} />
                            <a href="mailto:contact.tyche.team@gmail.com" className="email-address">
                                <span>contact.tyche.team@gmail.com</span>
                            </a>
                        </div>
                    </div>
                </div>
            </div>
            <div className="copy-right">۱۴۰۰-۱۴۰۱ کپی‌رایت © تمامی حقوق محفوظ است</div>
        </div>
    );
}
export default memo(Footer);