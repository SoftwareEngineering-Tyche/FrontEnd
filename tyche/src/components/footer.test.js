import { render } from "@testing-library/react"
import footer from "./footer";

test('renders the landing page', () => {
  render(<footer />);

  const { asFragment } = render(<footer text='Hello!' />);
  expect(asFragment()).toMatchSnapshot();
});
