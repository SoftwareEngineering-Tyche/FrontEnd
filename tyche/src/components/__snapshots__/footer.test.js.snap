// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the landing page 1`] = `
<DocumentFragment>
  <footer
    text="Hello!"
  />
</DocumentFragment>
`;
